src/Processors/Formats/Impl/DWARFBlockInputFormat.cpp

```diff
@@ -0,0 +1,792 @@
+                {
+                    /// We expect that this attribute appears before any attributes that point into the filename table.
+                    auto offset = val.getAsSectionOffset();
+                    if (offset.has_value())
+                        parseFilenameTable(unit, offset.value());
+                }
+
+                switch (val.getForm()) // (may be different from attr.Form because of DW_FORM_indirect)
+                {
+                    /// A 64-bit value.
+                    case llvm::dwarf::DW_FORM_data2:
+                    case llvm::dwarf::DW_FORM_data4:
+                    case llvm::dwarf::DW_FORM_data8:
+                    case llvm::dwarf::DW_FORM_data1:
+                    case llvm::dwarf::DW_FORM_sdata:
+                    case llvm::dwarf::DW_FORM_udata:
+                    case llvm::dwarf::DW_FORM_data16:
+                    case llvm::dwarf::DW_FORM_flag:
+                    case llvm::dwarf::DW_FORM_flag_present:
+                    case llvm::dwarf::DW_FORM_loclistx: // points to .debug_loclists
+                    case llvm::dwarf::DW_FORM_rnglistx: // points to .debug_rnglists
+                    case llvm::dwarf::DW_FORM_sec_offset: // points to some other section, depending on attr.Attr
+                    case llvm::dwarf::DW_FORM_implicit_const:
+                        if (need[COL_ATTR_INT]) col_attr_int->insertValue(val.getRawUValue());
+
+                        if (attr.Attr == llvm::dwarf::DW_AT_decl_line && std::exchange(need_decl_line, false))
+                            col_decl_line->insertValue(static_cast<UInt32>(val.getRawUValue()));
+
+                        /// Some attribute values are indices into lookup tables that we can stringify usefully.
+                        if ((attr.Attr == llvm::dwarf::DW_AT_decl_file || attr.Attr == llvm::dwarf::DW_AT_call_file) &&
+                            val.getRawUValue() < unit.filename_table_size) // filename
+                        {
+                            UInt64 idx = val.getRawUValue() + 1;
+                            if (attr.Attr == llvm::dwarf::DW_AT_decl_file && std::exchange(need_decl_file, false))
+                                col_decl_file.insertPosition(idx);
+
+                            if (need[COL_ATTR_STR])
+                            {
+                                auto data = unit.filename_table->getDataAt(idx);
+                                col_attr_str->insertData(data.data, data.size);
+                            }
+                        }
+                        else if (need[COL_ATTR_STR])
+                        {
+                            if (attr.Attr == llvm::dwarf::DW_AT_language) // programming language
+                                append(col_attr_str, removePrefix(llvm::dwarf::LanguageString(static_cast<uint32_t>(val.getRawUValue())),
+                                    strlen("DW_LANG_")));
```

Comment:
does it make sense to create `static constexpr std::string_view` for these common prefixes so you can simply call `.size()` method on them?

Reply:
Oh, nice trick, `constexpr size_t x = strlen("foo")` doesn't work (lame) but `constexpr std::string_view x = "foo"; ... x.size()` does.